Refreshing wrangler docs for 2025 (#1434)
ebi-ait/hca-ebi-wrangler-central: docs/SOPs/Introduction/dataset_wrangling_SOP.md

## Background
{: .no_toc }

This SOP document aims to describe the process of wrangling datasets into ingest for submission into the Data Portal.<br>
Keep track of the wrangling process documentation using the [wrangling process diagram](https://github.com/ebi-ait/hca-ebi-wrangler-central/blob/master/docs/SOPs/Introduction/wrangling_process_diagram.md).

We'll focus on submitting from a publication, but more alternatives are found [here](https://ebi-ait.github.io/hca-ebi-wrangler-central//SOPs/Introduction/submission_variations.html).

## Table of contents
{: .no_toc .text-delta }

1. TOC
{:toc}

Once a dataset needs wrangling is assigned to us, first thing to do is to decide on the consent type of the donors.

## Consent type

Before any metadata and data can be received from contributors they are required to sign the HCA Data contributor agreement (DCA). Two versions are available, 

> * a light DCA to be used for open access dataseta, requiring no institutional signer

Reviewer: ```suggestion
* a light DCA to be used for open access dataseta, requiring no institutional signer - _DCA is being finalised, todo: add link to pdf_
```

* a [full DCA](https://ebi-ait.github.io/hca-ebi-wrangler-central/SOPs/Managed_access/Data_Contributor_Agreement_explained.html) to be used for managed access dataset.

After confirming what type of consent the contributor have obtained for their dataset, send them the appropriate DCA. 

For more information on managed access SOPs, including how to sign the DCA read here:
* [DCA explained](https://ebi-ait.github.io/hca-ebi-wrangler-central/SOPs/docs/SOPs/Managed_access/Data_Contributor_Agreement_explained.html)
* [DCA SOP](https://ebi-ait.github.io/hca-ebi-wrangler-central/SOPs/Managed_access/Data_Contributor_Agreement_SOP.html)
* [Data Transfer SOP](https://ebi-ait.github.io/hca-ebi-wrangler-central/SOPs/Managed_access/Data_Transfer_SOP.html)
* [Pre-submission SOP](https://ebi-ait.github.io/hca-ebi-wrangler-central/SOPs/Managed_access/Pre-submission_SOP.html)

> * [Review and export SOP](https://ebi-ait.github.io/hca-ebi-wrangler-central/SOPs/Managed_access/Review_and_export_SOP.html)

Author: of course! applied in f22e8c36b933c6b61f38d0292f7b2a288142e52f

- SRA/ NCBI
    - [cloud delivery SOP](https://ebi-ait.github.io/hca-ebi-wrangler-central/SOPs/Access_data_files/NCBI_SRA_cold_storage_how_to.html)

**GEO Matrix data files:**
For most publications with a GEO accession, gene matrices files are available for download. The matrices files can be directly downloaded either locally to your desktop by clicking the link, or via wget in the terminal and on EC2.

There is an [SOP](https://ebi-ait.github.io/hca-ebi-wrangler-central/tools/Download_from_archives/download_GEO_matrices.html) to help with programmatic downloading of GEO supplementary files.

**Matrix and Cell types:**
For each project, wranglers should endeavour to find an expression matrix and if not embedded within that matrix, a listing of cell type annotations. These are generally linked from a publication, present as a supplementary file on the publication, GEO or ArrayExpress submission.

The preferred formats for matrix files are:
* `loom`
* `h5ad`
* `rds` or `Robj`
* cell ranger format (`csv` & `mtx`) - split into three files for `_features.csv`, `_matrix.mtx` and cell `_barcodes.csv`

> If analysis files are separated in count matrix/ feature/ barcode, join them into a single process to represent a single cell ranger run.

Reviewer: I would move this comment to the metadata section, here it would be confusing for someone who's not already familiar with the wrangling process

**Other repos:**
It is very common for contributors to upload their processed data into a custom webside that is not an archive. In this case, if you are sure there is consent to download and re-distribute data, you should get files locally.

## Data upload

Once all files are downloaded locally (or in EC2 / S3), the data files need to be uploaded to an `hca-util` upload area. This is an intermediate bucket that allows us fast upload to final upload area in case submission envelope changes.

You can follow the [upload instructions](https://github.com/ebi-ait/hca-documentation/wiki/How-to-upload-data-to-an-upload-area-using-hca-util).

### Data upload - contributor

> If contributor is to upload their data, this should be done via `hca-util` as well. In order to do that an aws account needs to be created using this [SOP](https://ebi-ait.github.io/hca-ebi-wrangler-central/SOPs/Access_data_files/aws_contributor_credentials.html).

Reviewer: ```suggestion
If contributor is providing their data, this should be done via `hca-util` as well. In order to do that an aws account needs to be created using this [SOP](https://ebi-ait.github.io/hca-ebi-wrangler-central/SOPs/Access_data_files/aws_contributor_credentials.html).
```

1. Tier 1 submission
1. Tier 2 submission
1. Submission from Published project
1. Submission from Unpublished project

### Tier-ed metadata

Integration teams are using their own distinct metadata schema, to accomodate their needs. We can [convert](https://github.com/ebi-ait/hca-tier1-to-dcp) this Tier-ed schema into the [HCA metadata schema](https://github.com/HumanCellAtlas/metadata-schema/tree/master/json_schema), in order to proceed with ingestion into the DCP2.0. 

More information can be found in the [confluence page](https://embl.atlassian.net/wiki/spaces/HCA/pages/142936546/HCA+metadata+schema+s+-+decisions).

Since Tier-ed metadata can be easily converted to HCA schema there is limited curation required for the submission to be complete. 

On the other hand, in cases 3-4 we would need to gather the metadata (case 3), or help the contributor to gather the metadata (case 4). In this SOP we will focus on the cases 3-4.

## Gathering data

> The first step in the wrangling process is always to make sure that we have files available to deposit. If no file can be associated with metadata, the submission will be invalid.

Author: Is it clearer in  522ae7fa2d31298054f4703ddd28168f72766e06 ?

## Gathering data

> After we are made sure that a DCA is in place, we can start with making sure that we have the files available to deposit. If no file can be associated with metadata, submission will not be possible to be ingested.

Reviewer: ```suggestion
Collecting data files is important because submissions with no data are not allowed in the Data portal, and a submission with no data will fail the validation step when ingested.
```

**Library preparation protocol** <br>
You can refer to the [assay cheat sheet](https://docs.google.com/spreadsheets/d/1H9i1BK-VOXtMgGVv8LJZZZ9rbTG4XCQTBRxErdqMvWk/edit#gid=0) for existing, standard assays so that we stay consistent across Lattice, EBI and UCSC wrangling teams.

### Filling in metadata about the files

For datasets with large number of files, the [ENA filename extractor tool](https://github.com/ebi-ait/hca-ebi-wrangler-central/blob/master/src/ena_filename_extractor.py) can be of use. It requires at least to have already filled the 'INSDC Experiment Accesion' at the 'Cell suspension' and the 'Sequence file' tabs. The wrangler has to manually download a JSON report from the corresponding project's page at ENA. This script will fill in the 'File name' column at the 'Sequence file' tab. 

For each expression matrix or cell type annotation file that is found, a row needs to be filled in the metadata spreadsheet, in the ‘Analysis file’ tab. Analysis files can be linked to biomaterial entities via processes. Information related to the analysis protocol is captured in the Analysis_protocol entity (See the Analysis protocol tab) linked to the process.

> ??? The best practice is to link the analysis files to sequence file entities, if possible.<br>

Reviewer: This sounds like a very old practice, it would be best to remove it from here
